### PR TITLE
Example of non-valid transaction because of ACL violation

### PR DESCRIPTION
**Pull request by tohar-ayash**
Signed-off-by: tohar-ayash <toharayash1@gmail.com>

### CODE REVIEW

File: examples/api/acl_violation/acl_violation.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	fmt.Println("Alice - adding key, value: key1, val1 to the database")
+	err = aliceTx.Put("db", "key1", []byte("val1"), nil)
+	if err != nil {
+		fmt.Printf("Adding new key to database failed, reason: %s\n", err.Error())
+		return
+	}
+
+	fmt.Println("Committing alice transaction")
+	txID, aliceReceipt, err := aliceTx.Commit(true)
+	if err != nil {
```
Comment (gennadylaventman) [inline examples/api/acl_violation/acl_violation.go:181]: First `if` may present execution of second `if`. Join both ifs together.

File: examples/api/acl_violation/acl_violation.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+		return
+	}
+
+	fmt.Println("Committing transaction")
+	txID, _, err = tx.Commit(true)
+	if err != nil {
+		fmt.Printf("Commit failed, reason: %s\n", err.Error())
+		return
+	}
+	fmt.Printf("Transaction number %s committed successfully\n", txID)
```
Comment (gennadylaventman) [inline examples/api/acl_violation/acl_violation.go:147]: Please add at least database creation phase to separate function. In addition, you can even create utility function `addUser()` with arguments as admin session, user name, location of certificate, list of names of dbs with read only permission and list of dbs with read-write permissions.

**Review by gennadylaventman — CHANGES_REQUESTED** [review]

File: examples/api/acl_violation/acl_violation.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	fmt.Println("Bob - adding key, value: key2, val2 to the database")
+	err = bobTx.Put("db", "key2", []byte("val2"), nil)
+	if err != nil {
+		fmt.Printf("Adding new key to database failed, reason: %s\n", err.Error())
+		return
+	}
+
+	fmt.Println("Committing bob transaction")
+	txID, bobReceipt, err := bobTx.Commit(true)
+	if err != nil {
```
Comment (gennadylaventman) [inline examples/api/acl_violation/acl_violation.go:221]: Same here - first `if` prevents second `if` execution.

File: examples/api/acl_violation/acl_violation.go (hunk 1 above)
Comment (tohar-ayash) [reply examples/api/acl_violation/acl_violation.go:73]: done

File: examples/api/acl_violation/acl_violation.go (hunk 2 above)
Comment (tohar-ayash) [reply examples/api/acl_violation/acl_violation.go]: done

File: examples/api/acl_violation/acl_violation.go (hunk 3 above)
Comment (tohar-ayash) [reply examples/api/acl_violation/acl_violation.go:113]: done

**Review by gennadylaventman — APPROVED** [review]

**Pull request merged**